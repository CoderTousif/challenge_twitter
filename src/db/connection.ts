import Knex from 'knex'
import KnexTinyLogger from 'knex-tiny-logger'

const config = require('../../knexfile')[process.env.NODE_ENV || 'development']

const knex =
  process.env.NODE_ENV === 'development'
    ? KnexTinyLogger(Knex(config))
    : Knex(config)

export default knex
