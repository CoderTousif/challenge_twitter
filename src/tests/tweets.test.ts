import db from '../db/connection'
import { FEED } from './queries/tweets.queries'
import { testClient } from './setup'

describe('Tweets', () => {
  beforeEach(async () => {
    await db.migrate.rollback()
    await db.migrate.latest()
    await db.seed.run()
  })

  afterEach(async () => {
    await db.migrate.rollback()
  })

  test('it should fetch the tweets with user', async () => {
    const { query } = await testClient()

    const res = await query({
      query: FEED,
    })

    expect(res.data.feed[0]).toMatchSnapshot()
  })
})
