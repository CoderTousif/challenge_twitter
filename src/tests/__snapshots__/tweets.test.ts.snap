// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweets it should fetch the tweets with user 1`] = `
Object {
  "body": "Quam repellat quaerat.",
  "id": "1",
  "user": Object {
    "display_name": "Violet",
    "id": "1",
    "username": "Casandra.Pouros31",
  },
  "visibility": "followers",
}
`;
